core/src/main/java/org/springframework/security/authentication/AuthenticationObservationConvention.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authentication;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationConvention;
+
+public final class AuthenticationObservationConvention
+		implements ObservationConvention<AuthenticationObservationContext> {
+
+	@Override
+	public KeyValues getLowCardinalityKeyValues(AuthenticationObservationContext context) {
+		KeyValues kvs = KeyValues.empty();
+		if (context.getAuthenticationRequest() != null) {
+			kvs = kvs.and("authentication.request.type", context.getAuthenticationRequest().getAuthenticationType());
```

Comment:
I would recommend adding a default "unknown", "n/a", "empty", "null", etc. value if the data is not available.

The reason behind this, is that that way you will always have the same set of tags on all of your observations so it will be easier for the users to navigate and also some metrics backends (e.g.: Prometheus) don't like if the number of tags are different.

So this would look like:
```java
if (context.getAuthenticationRequest() != null) {
    kvs = kvs.and("authentication.request.type", context.getAuthenticationRequest().getAuthenticationType());
}
else {
    kvs = kvs.and("authentication.request.type", "n/a");
}
```
or
```java
String requestType = context.getAuthenticationRequest() != null ? context.getAuthenticationRequest().getAuthenticationType() : "n/a";
kvs = kvs.and("authentication.request.type", requestType);
```